src/Avalonia.Controls.TreeDataGrid/Selection/TreeDataGridRowSelectionModel.cs

```diff
@@ -184,6 +188,85 @@ private void Search(TreeDataGrid treeDataGrid, string candidatePattern, int sele
             }
         }
 
+        void ITreeDataGridSelectionInteraction.OnPreviewKeyDown(TreeDataGrid sender, KeyEventArgs e)
+        {
+            static bool IsElementFullyVisibleToUser(TransformedBounds controlBounds)
+            {
+                return controlBounds.Clip.Contains(controlBounds.Bounds.TransformToAABB(controlBounds.Transform));
+            }
+            if (e.Key == Key.PageDown || e.Key == Key.PageUp)
+            {
+                var children = sender.RowsPresenter.GetLogicalChildren();
+                var childrenCount = children.Count();
```

Comment:
When testing this, I found that when e.g. scrolling up a page, more than a page of elements was scrolled at a time. I think this is because `children.Count` here can include non-realized elements. I think we should be using `sender.RowsPresenter.RealizedElements` instead of the logical children.